### ISSUE TITLE
Create 'ground truth seeding' dataset for validation interface

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#1445]
@maubinle has previously talked about 'ground truth seeding' (aka 'golden insertion') for our validation interfaces. I think this is a good idea. We need to figure out:

- How to implement this (from a code architecture and software engineering perspective). :)
- What are seeding dataset should be (mixture of all label types, which could be pulled from any city that we deploy in tbh but ideally should be per city probably otherwise might be noticeable)
- What our ground truth seeding rate should be. For example, we currently have validation missions that include 10 validations--should we include one ground truth seed in each batch (10% seed rate)? I think a rate of 10-20% is reasonable for maybe the first 100 validations and then the rate can lower (based on user performance)
- Also, and I think this is crucial, in some (or maybe all) cases with ground truth seeding, if the user gets it wrong, we should use this as a 'teachable moment' and explain to them what the right answer is (and why). This will take more time to implement.

See 'Ground Truth Seeding' sub-section in this crowdsourcing survey paper: http://crowdsourcing-class.org/readings/downloads/intro/QuinnAndBederson.pdf.

**Comment by maubinle** [comment]
This was a paper I was particularly impressed with that used a ground truth seeding technique: [crowdsourcing annotations for visual object detection](http://vision.stanford.edu/pdf/bbox_submission.pdf)

They use a 5-10% ground truth rate, so I think given our mission structure 10% does make sense. I do think the earlier we implement this, the more data we'll have to work with to give us some idea of "worker quality."

I've seen some papers where if the user "fails" the ground truth validation, their labels are entirely thrown out. This is a bit extreme imo, but maybe worth consideration.

**Comment by jonfroehlich** [comment]
Yes, 10% seems good. To create the seeding dataset, maybe we do majority vote from three researcher validations? How many total validations do we need? 100?

**Comment by jonfroehlich** [comment]
I'd like @aileenzeng to think about this more as part of her independent study. I think the rollout could be in two stages.

Stage One: simply implement a system where 1/10 validations have a known correct answer. I think for the 'known' correct answer, we could just pull on any labels validated already by someone marked as 'researcher' in the backend. (We could, potentially, be more rigorous about this and use any labels where two researchers have validated and agreed on their validation answer).

Stage Two: this will be more time consuming to design and implement and will require a curated set of validated labels. Setup a small set (maybe ~20-30) of carefully chosen validated labels (maybe ~5 per label type) that contain common mistakes. If the user gets it wrong, use this to help teach them about the right answer. So, for example, we could have 'driveways' and 'driveway transitions' examples labeled as curb ramps, which the user should mark as 'Disagree.' If they don't, we should teach them why not.

**Comment by jonfroehlich** [comment]
I think this is still an important idea and one that we should likely implement, especially when we integrate validation missions into the core user experience (i.e., when validation missions are given intermittently between labeling missions). Why?

Because, imo, the very act of doing validations helps train users as it shows them what others believe to be problems. But more concretely, with the golden insertions, we will actually train users by correcting them when they make a mistake in validating.

**Comment by jonfroehlich** [comment]
@misaugstad, @devonsnyder, and Yochai talked about this today and agreed that this should be a priority for us.

CC @crescendochu as it relates to interventions.

**Comment by jonfroehlich** [comment]
@misaugstad and I just discussed this:

* We only intervene if someone gets it wrong (not if they get it right)
* We intervene immediately when they get it wrong by explaining why/how they got it wrong (need to dev the UI for this)
* Do we need to customize per city (probably eventually but maybe not for the MVP--it would be quite obvious, however, if you are in Zurich or Mexico and suddenly see a Seattle image—maybe it's per country?).

**Comment by misaugstad** [comment]
We also talked about possibly forcing users to do only validations until they get enough of these correct before letting them move on to Explore page. Maybe we do this only in certain contexts, like service learning, etc?

**Comment by jonfroehlich** [comment]
> We also talked about possibly forcing users to do only validations until they get enough of these correct before letting them move on to Explore page. Maybe we do this only in certain contexts, like service learning, etc?

Right, thanks for writing this down. I think we may want to make a separate ticket for this as it does feel a bit orthogonal from whether and how we implement ground truth seeding. And I'd prioritize this current ticket (#1445).
